extensions/binding/org.eclipse.smarthome.binding.dmx/src/main/java/org/eclipse/smarthome/binding/dmx/handler/DimmerThingHandler.java

```diff
@@ -0,0 +1,286 @@
+            }
+            if (configuration.get(CONFIG_DIMMER_TURNONVALUE) != null) {
+                String turnOnValueString = String.valueOf(fadeTime) + ":"
+                        + ((String) configuration.get(CONFIG_DIMMER_TURNONVALUE)) + ":-1";
+                ValueSet turnOnValue = ValueSet.fromString(turnOnValueString);
+                if (turnOnValue != null) {
+                    this.turnOnValue = turnOnValue;
+                    logger.trace("set turnonvalue to {} in {}", turnOnValue, this.thing.getUID());
+                } else {
+                    logger.warn("could not set turnonvalue to {} in {}, malformed", turnOnValueString,
+                            this.thing.getUID());
+                }
+            }
+            if (configuration.get(CONFIG_DIMMER_TURNOFFVALUE) != null) {
+                String turnOffValueString = String.valueOf(fadeTime) + ":"
+                        + ((String) configuration.get(CONFIG_DIMMER_TURNOFFVALUE)) + ":-1";
+                ValueSet turnOffValue = ValueSet.fromString(turnOffValueString);
+                if (turnOffValue != null) {
+                    this.turnOffValue = turnOffValue;
+                    logger.trace("set turnoffvalue to {} in {}", turnOffValue, this.thing.getUID());
+                } else {
+                    logger.warn("could not set turnoffvalue to {} in {}, malformed", turnOffValueString,
+                            this.thing.getUID());
+                }
+            }
+            if (this.getBridge().getStatus().equals(ThingStatus.ONLINE)) {
+                updateStatus(ThingStatus.ONLINE);
+            } else {
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.BRIDGE_OFFLINE);
+            }
+        } else {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
+                    "DMX channel configuration missing in " + this.thing.getUID());
+        }
+    }
+
+    @Override
+    public void channelLinked(ChannelUID channelUID) {
+        switch (channelUID.getId()) {
+            case CHANNEL_SWITCH:
+                if (channels.size() > 0) {
+                    channels.get(0).addListener(channelUID, this, ListenerType.ONOFF);
+                }
+                break;
+            case CHANNEL_BRIGHTNESS:
+                if (channels.size() > 0) {
+                    channels.get(0).addListener(channelUID, this, ListenerType.VALUE);
+                }
+                break;
+            case CHANNEL_COLOR:
+                if (channels.size() > 2) {
+                    channels.get(0).addListener(channelUID, this, ListenerType.VALUE);
+                    channels.get(1).addListener(channelUID, this, ListenerType.VALUE);
+                    channels.get(2).addListener(channelUID, this, ListenerType.VALUE);
+                } else {
+                    logger.warn("cannot link color channel with less than three DMX channels in thing {}",
```

Comment:
This is DMX channels, not thing channels. Maybe I should refactor that, it might be confusing. Look at the latest DimmerThingHandler. java, l. 166-185 and l. 235-240.